editor/src/clj/editor/sprite.clj

```diff
@@ -453,91 +496,111 @@
+                        prop {:node-id _node-id
+                              :type property-type
+                              :edit-type edit-type
+                              :label label
+                              :value value
+                              :error error}]
+                    ;; Insert the original material values as original-value if there is a vertex override.
+                    (if (some? override-values)
+                      [property-key (assoc prop :original-value material-values)]
+                      [property-key prop]))))
+              material-attribute-infos)]
     (-> _declared-properties
         (update :properties into attribute-properties)
-        (update :display-order into attribute-property-keys))))
-
-(g/defnk produce-vertex-attribute-bytes [material-attribute-infos vertex-attribute-overrides]
-  (into {}
-        (map (fn [material-attribute-info]
-               (let [name-key (:name-key material-attribute-info)
-                     bytes (if-some [override-values (get vertex-attribute-overrides name-key)]
-                             (.array (graphics/make-attribute-byte-buffer (:data-type material-attribute-info) override-values))
-                             (:bytes material-attribute-info))]
-                 [name-key bytes])))
-        material-attribute-infos))
+        (update :display-order into (map first) attribute-properties))))
+
+(g/defnk produce-vertex-attribute-bytes [_node-id material-attribute-infos vertex-attribute-overrides]
+  (let [vertex-attribute-bytes
+        (into {}
+              (map (fn [{:keys [name-key] :as attribute-info}]
+                     (let [override-values (get vertex-attribute-overrides name-key)
+                           [bytes error] (if (nil? override-values)
+                                           [(:bytes attribute-info) (:error attribute-info)]
+                                           (let [{:keys [element-count semantic-type]} attribute-info
+                                                 resized-values (graphics/resize-doubles override-values semantic-type element-count)
+                                                 [bytes error-message :as bytes+error-message] (graphics/attribute->bytes+error-message attribute-info resized-values)]
+                                             (if (nil? error-message)
+                                               bytes+error-message
+                                               (let [property-key (attribute-key->property-key name-key)
+                                                     error (g/->error _node-id property-key :fatal override-values error-message)]
+                                                 [bytes error]))))]
+                       [name-key (or error bytes)])))
+              material-attribute-infos)]
+    (g/precluding-errors (vals vertex-attribute-bytes)
+      vertex-attribute-bytes)))
```

Comment:
Again, we can fail to encode the values and need to propagate the errors to elsewhere somehow.